nornir_nautobot/plugins/tasks/dispatcher/mikrotik_routeros.py

```diff
@@ -0,0 +1,105 @@
+
+        This accounts for Mikrotik Router OS config scrubbing behavior since ROS >= 7.X.
+
+        Args:
+            task (Task): Nornir Task.
+            logger (NornirLogger): Custom NornirLogger object to reflect job results (via Nautobot Jobs) and Python logger.
+            obj (Device): A Nautobot Device Django ORM object instance.
+            remove_lines (list): A list of regex lines to remove configurations.
+            substitute_lines (list): A list of dictionaries with to remove and replace lines.
+
+        Returns:
+            Result: Nornir Result object with a dict as a result containing the running configuration
+                { "config: <running configuration> }
+        """
+        task.host.platform = NETMIKO_DEVICE_TYPE
+        logger.log_debug(f"Analyzing Software Version for {task.host.name} on {task.host.platform}")
+        command = GET_VERSION_COMMAND
+        try:
+            result = task.run(task=netmiko_send_command, command_string=GET_VERSION_COMMAND)
+        except NornirSubTaskError as exc:
+            if isinstance(exc.result.exception, NetmikoAuthenticationException):
+                logger.log_failure(obj, f"Failed with an authentication issue: `{exc.result.exception}`")
+                raise NornirNautobotException(  # pylint: disable=W0707
+                    f"Failed with an authentication issue: `{exc.result.exception}`"
+                )
+
+            if isinstance(exc.result.exception, NetmikoTimeoutException):
+                logger.log_failure(obj, f"Failed with a timeout issue. `{exc.result.exception}`")
+                raise NornirNautobotException(  # pylint: disable=W0707
+                    f"Failed with a timeout issue. `{exc.result.exception}`"
+                )
+
+            logger.log_failure(obj, f"Failed with an unknown issue. `{exc.result.exception}`")
+            raise NornirNautobotException(  # pylint: disable=W0707
+                f"Failed with an unknown issue. `{exc.result.exception}`"
+            )
+
+        if result[0].failed:
+            return result
+
+        major_version = result[0].result.split()[3].split(".")[0]
+
+        command = GET_CONFIG_COMMAND
+        if major_version > "6":
+            command += " show-sensitive"
+
+        logger.log_debug(f"Found Mikrotik Router OS version {major_version}")
+        logger.log_debug(f"Executing get_config for {task.host.name} on {task.host.platform}")
+
+        try:
+            result = task.run(task=netmiko_send_command, command_string=command)
+        except NornirSubTaskError as exc:
+            logger.log_failure(obj, f"Failed with an unknown issue. `{exc.result.exception}`")
+            raise NornirNautobotException(  # pylint: disable=W0707
+                f"Failed with an unknown issue. `{exc.result.exception}`"
+            )
+
+        if result[0].failed:
+            return result
+
+        running_config = result[0].result
+
+        if remove_lines:
```

Comment:
I think it makes sense centralize some of this. I would make something like the below on the `NautobotNornirDriver`:

``` python
    @staticmethod
    def _remove_lines(<proper_args>):

        if remove_lines:
            logger.log_debug("Removing lines from configuration based on `remove_lines` definition")
            running_config = clean_config(running_config, remove_lines)

    @staticmethod
    def _substitute_lines(<proper_args>):


        if substitute_lines:
            logger.log_debug("Substitute lines from configuration based on `substitute_lines` definition")
            running_config = sanitize_config(running_config, substitute_lines)

    @staticmethod
    def _create_file(<proper_args>):

        make_folder(os.path.dirname(backup_file))

        with open(backup_file, "w", encoding="utf8") as filehandler:
            filehandler.write(running_config)
```

Then call it here with something like:

``` python
        self._remove_lines()
        self._substitute_lines()
        self._create_file()
        return Result(host=task.host, result={"config": running_config})
```

Reply:
would it be ok to break the pattern and make get_config(...) a normal Instance method? as static methods do not receive the instance as an arg. It would end up something like this:

`       NautobotNornirDriver._remove_lines(remove_lines: list, _running_config: str, logger)
        NautobotNornirDriver._substitute_lines(substitute_lines: list, _running_config: str, logger)
        NautobotNornirDriver._create_file(backup_file: str)`

Other option would be to convert get_config() into a classmethod, but that would also break the pattern used so far.

Reply:
`a normal Instance method`, I don;t think so, I think that will break the dispatcher.

Reply:
we can call them as class static methods as a short-term solution. Since that functionality is also in use by  `NornirNautobotDriver` and `NetmikoNautobotNornirDriver` I think it would make sense to incorporate them into "helpers":
How does that sound @itdependsnetworks ?

Reply:
What are helpers?

Reply:
nornir_nautobot/utils/helpers.py `make folder` resides there, I believe removelines(), substitutelines() and savefile() could be moved there as well so they can be reused.

Reply:
@itdependsnetworks changes implemented.